test/groovy/com/sap/piper/MtaUtilsTest.groovy

```diff
@@ -0,0 +1,247 @@
+        mtaUtils.generateMtaDescriptorFromPackageJson(srcPackageJson, '', 'testApplicationName')
+    }
+
+    @Test
+    void testTargetMtaDescriptorNull() {
+        thrown.expect(IllegalArgumentException)
+        thrown.expectMessage("The parameter 'targetMtaDescriptor' can not be null or empty.")
+        mtaUtils.generateMtaDescriptorFromPackageJson(srcPackageJson, null, 'testApplicationName')
+    }
+
+    @Test
+    void testApplicationNameEmpty() {
+        thrown.expect(IllegalArgumentException)
+        thrown.expectMessage("The parameter 'applicationName' can not be null or empty.")
+        mtaUtils.generateMtaDescriptorFromPackageJson(srcPackageJson, targetMtaDescriptor, '')
+    }
+
+    @Test
+    void testApplicationNameNull() {
+        thrown.expect(IllegalArgumentException)
+        thrown.expectMessage("The parameter 'applicationName' can not be null or empty.")
+        mtaUtils.generateMtaDescriptorFromPackageJson(srcPackageJson, targetMtaDescriptor, null)
+    }
+
+    @Test
+    void testMissingNameInJson() {
+        badJson = tmp.newFile('missingName.json')
+        badJson.text = missingNameInJson()
+        badJson.dump()
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("'name' not set in the given package.json.")
+
+        mtaUtils.generateMtaDescriptorFromPackageJson(badJson.absolutePath, targetMtaDescriptor, 'testApplicationName')
+    }
+
+    @Test
+    void testEmptyNameInJson() {
+        badJson = tmp.newFile('emptyName.json')
+        badJson.text = emptyNameInJson()
+        badJson.dump()
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("'name' not set in the given package.json.")
+
+        mtaUtils.generateMtaDescriptorFromPackageJson(badJson.absolutePath, targetMtaDescriptor, 'testApplicationName')
+    }
+
+    @Test
+    void testMissingVersionInJson() {
+        badJson = tmp.newFile('missingVersion.json')
+        badJson.text = missingVersionInJson()
+        badJson.dump()
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("'version' not set in the given package.json.")
+
+        mtaUtils.generateMtaDescriptorFromPackageJson(badJson.absolutePath, targetMtaDescriptor, 'testApplicationName')
+    }
+
+    @Test
+    void testEmptyVersionInJson() {
+        badJson = tmp.newFile('emptyVersion.json')
+        badJson.text = emptyVersionInJson()
+        badJson.dump()
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("'version' not set in the given package.json.")
+
+        mtaUtils.generateMtaDescriptorFromPackageJson(badJson.absolutePath, targetMtaDescriptor, 'testApplicationName')
+    }
+
+    @Test
+    void testFileGenerated() {
+        mtaUtils.generateMtaDescriptorFromPackageJson(srcPackageJson, targetMtaDescriptor, 'testApplicationName')
+        def file = new File(targetMtaDescriptor)
+        assert file.exists()
+        assert file.canRead()
```

Comment:
```File#canRead()``` implies ```File#exists()``` -> ```File#exists()```check can be removed. By the way: I doubt that this test adds any real values - as already outline above in an other comment. This test only ensures that the closure registered for ```writeYaml``` works. It does not ensure that the file will be really written later on when the real code of ```writeYaml``` is triggered later in the free wild.

Reply:
reworked as suggested below